test/test.hs

```diff
@@ -2291,6 +2291,100 @@ tests = testGroup "hevm"
       assertEqualM "number of counterexamples" 1 numCexes
       assertEqualM "number of errors" 0 numErrs
       assertEqualM "number of qed-s" 0 numQeds
+    , test "copyslice-symbolic-ok" $ do
+      Just c <- solcRuntime "C"
+        [i|
+         contract Target {
+           function get(address addr) external view returns (uint256) {
+               return 55;
+           }
+         }
+         contract C {
+           function retFor(address addr) public returns (uint256) {
+               Target mm = new Target();
+               uint256 ret = mm.get(addr);
+               assert(ret == 4);
+               return ret;
+           }
+         }
+        |]
+      let sig2 = Just (Sig "retFor(address)" [AbiAddressType])
+      (expr, ret) <- withDefaultSolver $ \s -> checkAssert s defaultPanicCodes c sig2 [] defaultVeriOpts
+      putStrLnM $ "successfully explored: " <> show (Expr.numBranches expr) <> " paths"
+      assertBoolM "The expression is NOT error" $ not $ any isError ret
+    -- NOTE: below used to be symbolic copyslice copy error before new copyslice simplification:
+    --       go (CopySlice (Lit 0) (Lit 0) (BufLength (AbstractBuf k1))
+    --         (CopySlice (Lit 0) (Lit 0) (BufLength (AbstractBuf k2)) (AbstractBuf k3) _)
+    --         (ConcreteBuf "")) | k1 == k2 && k2 == k3 = (AbstractBuf k1)
+    , test "overapproximates-undeployed-contract" $ do
```

Comment:
Can this test be merged with the previous one?
It is doing the same work, no?
Only checks on the result are different.

Reply:
Actually, they are different, weirdly enough! The one above actually works without the simplifications. But this one does not. It's because Target was not deployed in this case, so it's abstracted over. Since we are not running a forge test, we don't have a setUp() and nothing is set up, so mm is actually not deployed. Weird, but true.

Reply:
Oh, I see now.
Yeah, that's pretty weird.
The state variable `mm` is never assigned.
Won't the variable have always value `0`? Shouldn't we that call `mm.get` is always trying to call function on address `0`?

Reply:
Oh good point, it is trying to call address 0. But there is nothing at 0, so it overapproximates. Which is OK in this case, I think. Although we could emit a warning for call to address zero. Let me create an issue to track this. Would be hepful to emit a warning probably. It can happen, but in this case, we can't really do much about it. However, it is likely unintended.

In the meanwhile, I'm merging, with a note that this is address 0, and the contract is obviously not deployed there...

Reply:
Documented here: https://github.com/ethereum/hevm/issues/671

Reply:
One last note, sorry for distracting you.
But, if we know the call is to zero address, we know that it will revert, no?

I don't know if I understand correctly how EVM works, but the call will lead to revert, so the assertion will not be reached, no?
It seems to me from your description that the counterexamples show how the assertion can be reached, but these are spurious, no?

Let me know if I am misunderstanding something.

Reply:
Yeah, good point. This is getting to be bigger than I wanted... can we merge this and document that as part of #671 ?

Reply:
It's getting to be too much to keep in mind and work on at the same time...

Reply:
OK!